### PR TITLE
allow individual routes to be excluded from authentication

### PR DESCRIPTION
**Pull request by cofin**
# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [x] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [x] Have you updated the reference documentation?

### CODE REVIEW

**Review by Goldziher — COMMENTED** [review]
Looks good. 

I'd suggest we mage the actual opt key configurable though. I.e. make it an optional constructor kwarg with a default value.

**Comment by cofin** [comment]
Is this what you meant, or where you actually thinking we should add an attribute to the config for which key to use?

**Comment by Goldziher** [comment]
No, I meant inside the Middleware have the opt key configurable

**Comment by Goldziher** [comment]
Interesting idea I had - allow passing in arbitrary kwargs in route handler decorators and SRT them inside the opt dict

**Comment by cofin** [comment]
> ```python
> @get("/", my_kwarg="1234")
> def handler() -> None:
>    ...
> 
> assert handler.opt.get("my_kwarg")
> ```

Does it already work like this?  If so, I had no idea!

File: starlite/handlers/asgi.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def __init__(
         path: Union[Optional[str], Optional[List[str]]] = None,
         *,
         exception_handlers: Optional[ExceptionHandlersMap] = None,
+        exclude_from_auth: Optional[bool] = False,
```
[the remark this answers is not in the document]
Comment (Goldziher) [reply starlite/handlers/asgi.py:22]: can you make this a change suggestion?

https://haacked.com/archive/2019/06/03/suggested-changes/

[the remark this answers is not in the document]
Comment (cofin) [reply starlite/handlers/asgi.py:22]: It should be a bool that defaults to False

[the remark this answers is not in the document]
Comment (Goldziher) [reply starlite/handlers/asgi.py:22]:
```suggestion
        exclude_from_auth: bool = False,
```

File: starlite/handlers/http.py [hunk 2]
```diff
@@ -250,6 +251,7 @@ def __init__(
         cache: Union[bool, int] = False,
         cache_key_builder: Optional[CacheKeyBuilder] = None,
         dependencies: Optional[Dict[str, Provide]] = None,
+        exclude_from_auth: Optional[bool] = False,
```
[the remark this answers is not in the document]
Comment (cofin) [reply starlite/handlers/http.py:254]: yeah, this was in there before you mode the opt changes from yesterday or the day before.  I'll refactor it accordingly now.

**Comment by cofin** [comment]
> 



> Interesting idea I had - allow passing in arbitrary kwargs in route handler decorators and srt them inside the opt dict.

Because we are able to do this now, the implementation looks the same and is the key is dynamic.  This is really useful.  

Take a look and see if this is ready to merge now.

**Review by Goldziher — APPROVED** [review]

**Pull request merged**